management/server/file_store.go

```diff
@@ -141,9 +136,52 @@ func (s *FileStore) AddAccount(account *Account) error {
 	return nil
 }
 
-// GetPeersForAPeer returns a list of peers available for a given peer (key)
+// GetAccountPeers returns a list of peers available for a given account
 // Effectively all the peers of the original peer's account if any
-func (s *FileStore) GetPeersForAPeer(peerKey string) ([]string, error) {
+func (s *FileStore) GetAccountPeers(accountId string) ([]*Peer, error) {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+
+	_, accountFound := s.Accounts[accountId]
+	if !accountFound {
+		return nil, status.Errorf(codes.Internal, "account not found")
+	}
+	var peers []*Peer
+	for _, peer := range s.Peers {
+		if peer.AccountId == accountId {
+			peers = append(peers, peer)
+		}
+	}
+
+	return peers, nil
+}
+
+func (s *FileStore) GetAccountBySetupKey(setupKey string) (*Account, error) {
+
+	accountId, accountIdFound := s.SetupKeyId2AccountId[strings.ToLower(setupKey)]
+	if !accountIdFound {
+		return nil, status.Errorf(codes.NotFound, "provided setup key doesn't exists")
+	}
+
+	account, err := s.GetAccount(accountId)
+	if err != nil {
+		return nil, err
+	}
+
+	return account, nil
+}
+
+func (s *FileStore) GetAccount(accountId string) (*Account, error) {
+
+	account, accountFound := s.Accounts[accountId]
+	if !accountFound {
+		return nil, status.Errorf(codes.Internal, "account not found")
```

Comment:
With Go, this function doesn't make a lot of sense because every call to it is exact function implementation anyway. Only error message is standard

```
	account, err := s.GetAccount(accountId)
	if err != nil {
		return nil, err
	}

	return account, nil
```

Reply:
Fixed.